pkg/metrics/scraper.go

```diff
@@ -203,3 +245,38 @@ func (ps *PrometheusScraper) GetCPUUtilizationBreachDataPoints(namespace,
 	}
 	return dataPoints, nil
 }
+
+func (ps *PrometheusScraper) GetPodReadyLatencyByWorkload(namespace string,
+	workload string,
+) (float64, error) {
+
+	ctx, cancel := context.WithTimeout(context.Background(), ps.queryTimeout)
+	defer cancel()
+
+	query := fmt.Sprintf("min((%s"+
+		"{namespace=\"%s\"} - on (namespace,pod) (%s{namespace=\"%s\"}))  * on (namespace,pod) group_left(workload, workload_type)"+
+		"(%s{namespace=\"%s\", workload=\"%s\","+
+		" workload_type=\"deployment\"}))",
+		ps.metricRegistry.podReadyTimeMetric,
+		namespace,
+		ps.metricRegistry.podCreatedTimeMetric,
+		namespace,
+		ps.metricRegistry.podOwnerMetric,
+		namespace,
+		workload)
+
+	result, _, err := ps.api.Query(ctx, query, time.Now())
+
+	if err != nil {
+		return 0.0, fmt.Errorf("failed to execute Prometheus query: %v", err)
+	}
+	if len(result.String()) == 0 {
+		return 0.0, fmt.Errorf("failed to get result from query: %v", err)
+	}
+	podBootstrapTime, err := strconv.ParseFloat(strings.Split(strings.Split(result.String(), "=> ")[1], " @")[0], 32)
```

Comment:
Why are we reading the result as a string. Please use the appropriate type from github.com/prometheus/common/model package. Do not deserialize it manually.